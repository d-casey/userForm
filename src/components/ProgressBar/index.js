import React from 'react'
import stageConfig from '../../config/stageConfig'
import PropTypes from 'prop-types'
import progressBar from '../../css/progressBar.css'

const ProgressBar = ({ appStage }) => {
  const stages = stageConfig.progressBarStages

  const listItems = stages.map((stage, index) =>
    <li key={stage} className={(index === appStage) ? 'progress-bar-flex-item active-item' : 'progress-bar-flex-item'}>
      {stage}
    </li>
  )

  return (
    <div className="progress-bar">
      <ul className="progress-bar-flex-container">
        {listItems}
      </ul>
    </div>
  )
}

ProgressBar.propTypes = {
  appStage: PropTypes.number
}

export default ProgressBar
