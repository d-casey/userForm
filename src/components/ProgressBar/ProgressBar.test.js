import React from 'react'
import { create } from 'react-test-renderer'
import ProgressBar from '.'

test('renders correctly when empty', () => {
  const component = create(<ProgressBar />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('renders the correct content for the initial stage', () => {
  const component = create(<ProgressBar appStage={0} />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('renders the correct content for the second stage', () => {
  const component = create(<ProgressBar appStage={1} />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('renders the correct content for the third stage', () => {
  const component = create(<ProgressBar appStage={2} />)
  expect(component.toJSON()).toMatchSnapshot()
})
