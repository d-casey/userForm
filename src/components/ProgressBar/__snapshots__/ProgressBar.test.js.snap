// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when empty 1`] = `
<div
  className="progress-bar"
>
  <ul
    className="progress-bar-flex-container"
  >
    <li
      className="progress-bar-flex-item"
    >
      User
    </li>
    <li
      className="progress-bar-flex-item"
    >
      Privacy
    </li>
    <li
      className="progress-bar-flex-item"
    >
      Done
    </li>
  </ul>
</div>
`;

exports[`renders the correct content for the initial stage 1`] = `
<div
  className="progress-bar"
>
  <ul
    className="progress-bar-flex-container"
  >
    <li
      className="progress-bar-flex-item active-item"
    >
      User
    </li>
    <li
      className="progress-bar-flex-item"
    >
      Privacy
    </li>
    <li
      className="progress-bar-flex-item"
    >
      Done
    </li>
  </ul>
</div>
`;

exports[`renders the correct content for the second stage 1`] = `
<div
  className="progress-bar"
>
  <ul
    className="progress-bar-flex-container"
  >
    <li
      className="progress-bar-flex-item"
    >
      User
    </li>
    <li
      className="progress-bar-flex-item active-item"
    >
      Privacy
    </li>
    <li
      className="progress-bar-flex-item"
    >
      Done
    </li>
  </ul>
</div>
`;

exports[`renders the correct content for the third stage 1`] = `
<div
  className="progress-bar"
>
  <ul
    className="progress-bar-flex-container"
  >
    <li
      className="progress-bar-flex-item"
    >
      User
    </li>
    <li
      className="progress-bar-flex-item"
    >
      Privacy
    </li>
    <li
      className="progress-bar-flex-item active-item"
    >
      Done
    </li>
  </ul>
</div>
`;
